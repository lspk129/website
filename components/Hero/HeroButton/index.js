import styled from 'styled-components';

const Button = styled.button`
  position: relative;
  display: block;
  margin: 30px auto;
  overflow: hidden;
  border-width: 0;
  outline: none;
  border-radius: 2px;
  box-shadow: 0 1px 4px rgba(0, 0, 0, .6);
  background-color: #ff6666;
  color: #fff;
  transition: background-color .3s;
  &:hover {
    background-color: #fff;
    color: #ff6666
  };
  padding: 15px 50px;
  top: 65%;
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
`;

export default Button;
