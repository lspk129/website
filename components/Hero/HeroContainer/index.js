import styled from 'styled-components';

const HeroContainer = styled.div`
  padding: 100px 0px;
  justify-self: center;
  text-align: center;
`;

export default HeroContainer;
