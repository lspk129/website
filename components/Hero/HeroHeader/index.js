import styled from 'styled-components';

const HeaderContainer = styled.div`
  grid-row: 1 / 2;
  align-text: center;
  justify-self: center;
`;

export default HeaderContainer;
