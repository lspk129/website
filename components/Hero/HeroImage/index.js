import styled from 'styled-components';

const HomeImage = styled.div`
  background-image: url("../../../static/img21.jpg");
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-size: cover;
  display: grid;
  grid-template-rows: 1fr 5fr;
  align-items: start;
  justify-items: center;
`;

export default HomeImage;
