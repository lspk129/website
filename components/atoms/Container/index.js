import styled from 'styled-components';

const Container = styled.div`
  ${({ display }) => display && `display: ${display}`}
  ${({ textAlign }) => textAlign && `text-align: ${textAlign}`};
  ${({ alignSelf }) => alignSelf && `align-self: ${alignSelf}`};
  ${({ alignItems }) => alignItems && `align-items: ${alignItems}`};

  /* margin properties */
  ${({ m }) => m && `margin: ${m}rem;`}
  ${({ mx }) => mx && `margin: 0rem ${mx}rem;`}
  ${({ my }) => my && `margin: ${my}rem 0rem;`}
  ${({ mt }) => mt && `margin-top: ${mt}rem;`}
  ${({ mb }) => mb && `margin-bottom: ${mb}rem;`}
  ${({ mr }) => mr && `margin-right: ${mr}rem;`}
  ${({ ml }) => ml && `margin-left: ${ml}rem;`}

  /* padding properties */
  ${({ p }) => p && `padding: ${p}rem;`}
  ${({ px }) => px && `padding: 0rem ${px}rem;`}
  ${({ py }) => py && `padding: ${py}rem 0rem;`}
  ${({ pt }) => pt && `padding-top: ${pt}rem;`}
  ${({ pb }) => pb && `padding-bottom: ${pb}rem;`}
  ${({ pr }) => pr && `padding-right: ${pr}rem;`}
  ${({ pl }) => pl && `padding-left: ${pl}rem;`}

  /* background properties */
  ${({ bg, src }) => bg && `
    background-image: url(${src});
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    display: grid;
  `}
  ${({ backgroundColor }) => backgroundColor && `
    background-color: ${backgroundColor};
  `}
  ${({ zIndex }) => zIndex && `z-index: ${zIndex};`}
`;

export default Container;
