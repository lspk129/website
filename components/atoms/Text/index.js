import { createElement } from 'react';
import styled from 'styled-components';

const Text = styled(({ tag, hero, upperCase, children, ...props }) => createElement(tag, props, children))`
  text-align: center;

  ${({ tag, theme }) => tag === 'a' && `
    text-decoration: none;
    cursor: pointer;
    color: inherit;
    &:hover { color: ${theme.main}}
  `}

  ${({ tag, hero, theme }) => tag === 'h1' && `
    ${hero && `
      color: ${theme.text};
      font-size: 4rem;
      white-space: nowrap;
      font-weight: bold;
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
      line-height: 4rem;
    `}
  `}

  ${({ tag, hero, theme }) => tag === 'h2' && `
    ${hero && `
      color: ${theme.text};
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.2);
    `}
  `}

  ${({ upperCase }) => upperCase && 'text-transform: uppercase'};
`;

Text.defaultProps = {
  tag: 'span',
};

export default Text;
