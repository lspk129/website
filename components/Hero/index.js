import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import HeroImage from './HeroImage';
import HeroContainer from './HeroContainer';
import HeroTitle from './HeroTitle';
import HeroSubtitle from './HeroSubtitle';

const Hero = () => (
  <HeroImage>
    <HeroContainer>
      <HeroTitle>
        Moka Haiku:<br />web app agency
      </HeroTitle>
      <HeroSubtitle>
        Creating fast & interactive web applications with React and Meteor
      </HeroSubtitle>
      <RaisedButton label="get a free consultation" secondary />
    </HeroContainer>
  </HeroImage>
);

export default Hero;
