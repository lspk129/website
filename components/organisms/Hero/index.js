import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import Division from '../../atoms/Division';
import Text from '../../atoms/Text';

const Hero = () => (
  <Division bg src={'../../../static/Hero_image.jpg'} py={4}>
    <Text tag={'h1'} hero>
      Moka Haiku:<br />web app agency
    </Text>
    <Text tag={'h2'} hero>
      Creating fast & interactive web applications with React and Meteor
    </Text>
    <Division textAlign={'center'} pt={2}>
      <RaisedButton label={'get a free consultation'} secondary />
    </Division>
  </Division>
);

export default Hero;
