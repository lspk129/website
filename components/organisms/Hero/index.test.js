import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import RaisedButton from 'material-ui/RaisedButton';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import injectTapEventPlugin from 'react-tap-event-plugin';

import Hero from './';
import Container from '../../atoms/Container';
import Text from '../../atoms/Text';

if (!process.tapEventInjected) {
  injectTapEventPlugin();
  process.tapEventInjected = true;
}

describe('Hero', () => {
  const muiTheme = getMuiTheme();
  const mountWithContext = node => mount(node, {
    context: { muiTheme },
    childContextTypes: { muiTheme: PropTypes.object },
  });
  const wrapper = mountWithContext(<Hero />);

  it('renders correctly', () => {
    expect(wrapper.find(Hero)).toBePresent();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Container component', () => {
    expect(wrapper.find(Container)).toBePresent();
  });

  it('renders Text component with content "Moka Haiku:web app agency"', () => {
    expect(wrapper.find(Text).at(0)).toBePresent();
    expect(wrapper.find(Text).at(0)).toHaveText('Moka Haiku:web app agency');
  });

  it('renders Text component with content "Creating fast & interactive web applications with React and Meteor"', () => {
    expect(wrapper.find(Text).at(1)).toBePresent();
    expect(wrapper.find(Text).at(1)).toHaveText('Creating fast & interactive web applications with React and Meteor');
  });

  it('renders RaisedButton component', () => {
    expect(wrapper.find(RaisedButton)).toBePresent();
    expect(wrapper.find(RaisedButton)).toHaveProp('label', 'get a free consultation');
  });
});

