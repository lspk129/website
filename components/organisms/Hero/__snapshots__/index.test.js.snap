// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<Hero>
  <styled.div
    bg={true}
    py={4}
    src="../../../static/Hero_image.jpg"
  >
    <div
      className="sc-bdVaJa jomBZL"
      src="../../../static/Hero_image.jpg"
    >
      <Styled(Component)
        hero={true}
        tag="h1"
      >
        <Component
          className="sc-bwzfXH kDeLAv"
          hero={true}
          tag="h1"
        >
          <h1
            className="sc-bwzfXH kDeLAv"
          >
            Moka Haiku:
            <br />
            web app agency
          </h1>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        hero={true}
        tag="h2"
      >
        <Component
          className="sc-bwzfXH iUDRwd"
          hero={true}
          tag="h2"
        >
          <h2
            className="sc-bwzfXH iUDRwd"
          >
            Creating fast & interactive web applications with React and Meteor
          </h2>
        </Component>
      </Styled(Component)>
      <styled.div
        pt={2}
        textAlign="center"
      >
        <div
          className="sc-bdVaJa hNJaJD"
        >
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="get a free consultation"
            labelPosition="after"
            primary={false}
            secondary={true}
          >
            <Paper
              circle={false}
              rounded={true}
              style={
                Object {
                  "display": "inline-block",
                  "minWidth": 88,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
              transitionEnabled={true}
              zDepth={1}
            >
              <div
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "backgroundColor": "#ffffff",
                    "borderRadius": 2,
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                           0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "minWidth": 88,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <EnhancedButton
                  containerElement="button"
                  disabled={false}
                  focusRippleColor="#ffffff"
                  focusRippleOpacity={0.16}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onKeyboardFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  onTouchTap={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff4081",
                      "borderRadius": 2,
                      "height": 36,
                      "lineHeight": "36px",
                      "padding": 0,
                      "textAlign": "center",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  touchRippleColor="#ffffff"
                  touchRippleOpacity={0.16}
                  type="button"
                >
                  <button
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    onTouchTap={[Function]}
                    style={
                      Object {
                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                        "backgroundColor": "#ff4081",
                        "border": 10,
                        "borderRadius": 2,
                        "boxSizing": "border-box",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontFamily": "Roboto, sans-serif",
                        "fontSize": "inherit",
                        "fontWeight": "inherit",
                        "height": 36,
                        "lineHeight": "36px",
                        "margin": 0,
                        "muiPrepared": true,
                        "outline": "none",
                        "padding": 0,
                        "position": "relative",
                        "textAlign": "center",
                        "textDecoration": "none",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "verticalAlign": null,
                        "width": "100%",
                        "zIndex": 1,
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <TouchRipple
                      abortOnScroll={true}
                      color="#ffffff"
                      opacity={0.16}
                    >
                      <div
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                      >
                        <div
                          style={
                            Object {
                              "backgroundColor": false,
                              "borderRadius": 2,
                              "height": 36,
                              "muiPrepared": true,
                              "top": 0,
                              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "color": "#ffffff",
                                "fontSize": 14,
                                "fontWeight": 500,
                                "letterSpacing": 0,
                                "margin": 0,
                                "muiPrepared": true,
                                "opacity": 1,
                                "paddingLeft": 16,
                                "paddingRight": 16,
                                "position": "relative",
                                "textTransform": "uppercase",
                                "userSelect": "none",
                              }
                            }
                          >
                            get a free consultation
                          </span>
                        </div>
                      </div>
                    </TouchRipple>
                  </button>
                </EnhancedButton>
              </div>
            </Paper>
          </RaisedButton>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Hero>
`;
